ion/bitstream.go

```diff
@@ -949,3 +958,17 @@ func (b *bitstack) pop() {
 	}
 	b.arr = b.arr[:len(b.arr)-1]
 }
+
+// The size of an ordered struct is in the byte after Type byte.
+func (b *bitstream) handleOrderedStruct() (uint64, error) {
+	c, err := b.read()
+	if err != nil {
+		return 0, err
+	}
+	if c == -1 {
+		b.code = bitcodeEOF
+		return 0, nil
+	}
+	length := c & 0x7F
```

Comment:
Here's the relevant bit of [the `struct` encoding spec](http://amzn.github.io/ion-docs/docs/binary.html#13-struct):

> When L is 1, the struct has at least one symbol/value pair, the length field exists, and the field name integers are sorted in increasing order.

The `length` field is a `VarUInt`, so we have to allow for the case where it takes more than one byte to represent it. We should use the `readVarUintLen` function instead.